spark/src/test/scala/ai/chronon/spark/test/AnalyzerTest.scala

```diff
@@ -338,6 +340,43 @@ class AnalyzerTest {
 
   }
 
+
+  @Test(expected = classOf[java.lang.AssertionError])
+  def testJoinAnalyzerInvalidTablePermissions(): Unit = {
+    val spark: SparkSession = SparkSessionBuilder.build("AnalyzerTest" + "_" + Random.alphanumeric.take(6).mkString, local = true)
+    val tableUtils = spy(TableUtils(spark))
+    when(tableUtils.checkTablePermission(any(), any())).thenReturn(false)
```

Comment:
consider mocking the `tableUtils.sql` to throw a runtime exception to mimic table permission, since the timestamp check logic will try to access the table data using `tableUtils.sql`, and we want to ensure that this is not triggered and gated by the permission check first.